src/lex.rs

```diff
@@ -1,185 +1,121 @@
-use std::{ops::Range, slice::SliceIndex, str::Chars};
+//! Wave lexer
 
-#[derive(Clone, Copy, Debug, PartialEq)]
-pub enum Token {
-    Whitespace,
+use std::fmt::Display;
+
+use logos::{Lexer, Logos, Span};
 
-    Dash,
+/// Represents a WAVE token.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Logos)]
+#[logos(error = LexingError)]
+#[logos(skip r"[ \t\r\n\f]+")]
+#[logos(subpattern label_word = r"[a-z][a-z0-9]*|[A-Z][A-Z0-9]*")]
```

Comment:
Why does the label_world pattern include upper-case letters? It's my understanding that Wit identifiers are only lower-case alphanumeric.

Reply:
Label words can be either all lowercase or all uppercase. I believe the idea is to allow retaining uppercase identifiers in bindings for languages that conventionally do that like Python.

https://github.com/WebAssembly/component-model/blob/main/design/mvp/Explainer.md
```ebnf
word          ::= [a-z] [0-9a-z]*
                | [A-Z] [0-9A-Z]*
```

Reply:
FWIW, this is what WAC implements now as well (added here https://github.com/peterhuene/wac/pull/29).